### PR TITLE
Stop adapted actorsystem after running TypedSpec (#22763)

### PR DESCRIPTION
**Pull request by raboof**
Refs #22763

### CODE REVIEW

File: akka-typed-tests/src/test/scala/akka/typed/TypedSpec.scala [hunk 1]
```diff
@@ -57,7 +57,7 @@ abstract class TypedSpec(val config: Config) extends TypedSpecSetup {
   private var adaptedSystemUsed = false
   lazy val adaptedSystem: ActorSystem[TypedSpec.Command] = {
     val sys = ActorSystem.adapter(AkkaSpec.getCallerName(classOf[TypedSpec]), guardian(), config = Some(config withFallback AkkaSpec.testConf))
-    adaptedSystemUsed = false
+    adaptedSystemUsed = true
```
Comment (patriknw) [inline akka-typed-tests/src/test/scala/akka/typed/TypedSpec.scala]: LOL

File: akka-typed-tests/src/test/scala/akka/typed/TypedSpec.scala [hunk 2]
```diff
@@ -202,13 +202,15 @@ class TypedSpecSpec extends TypedSpec {
...
+
       implicit def system: ActorSystem[TypedSpec.Command]
 
       def `must report failures`(): Unit = {
-        a[TestFailedException] must be thrownBy {
+        a[MyException] must be thrownBy {
           sync(runTest("failure")(StepWise[String]((ctx, startWith) ⇒
             startWith {
-              fail("expected")
+              throw new MyException("expected")
```
Comment (patriknw) [inline akka-typed-tests/src/test/scala/akka/typed/TypedSpec.scala]: +1

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-typed-tests/src/test/scala/akka/typed/TypedSpec.scala (hunk 2 above)
Comment (johanandren) [inline akka-typed-tests/src/test/scala/akka/typed/TypedSpec.scala]: Don't we have TE in scope even here?

Comment (raboof) [inline akka-typed-tests/src/test/scala/akka/typed/TypedSpec.scala]: TE?

Comment (johanandren) [inline akka-typed-tests/src/test/scala/akka/typed/TypedSpec.scala]: Oh, sorry, it's a prepared test exception, but it's only in the stream teskit I see now: https://github.com/akka/akka/blob/23ca48bdcca096170db3c396b762ff8dd3a91737/akka-stream-testkit/src/test/scala/akka/stream/testkit/Utils.scala

**Review by johanandren — APPROVED** [review]
LGTM

**Pull request merged**